qiita_db/study.py

```diff
@@ -347,6 +348,59 @@ def create(cls, owner, title, efo, info, investigation=None):
 
         return cls(study_id)
 
+    @classmethod
+    def delete(cls, id_):
+        r"""Deletes the study from the database
+
+        Parameters
+        ----------
+        id_ : integer
+            The object identifier
+
+        Raises
+        ------
+        QiitaDBError
+            If the sample_(id_) table exists means a sample template exists
+        """
+        cls._check_subclass()
+
+        # checking that the id_ exists
+        cls(id_)
```

Comment:
@antgonza and I discussed this offline. The `_check_id` method is not a `classmethod` so it cannot be called here.

As a note, I think this brings up the discussion I started in #1045, so we can avoid all these checks...